osu.Game/Screens/Select/Carousel/UpdateBeatmapSetButton.cs

```diff
@@ -88,6 +92,32 @@ private void load()
                 },
             });
 
+            beatmapDownloader.DownloadFailed += _ =>
+            {
+                Hide();
+
+                // Download might have failed because BeatmapSetOnlineAvailability is DownloadDisabled
+                // We want to check that so reset online info for now, background processing will update the info
```

Comment:
What about the case where a download failed due to internet connection loss. This should be guarded properly rather than forcing background reprocess on any failure. I would hope the API would return a readable error that can be parsed here to automatically hide "update" button from appearing.